### PR TITLE
python3.pkgs.fetchPypi: deprecate in favor of top-level fetchPypi

### LINKED ISSUE #233252: Remove usages of non-toplevel `fetchPypi`
Since #211256, `fetchPypi` was moved to top level, and `python3.pkgs.fetchPypi` and `python3Packages.fetchPypi` are aliases of `fetchPypi` from the top level

To find the usages of these aliases, you can search for `\.fetchPypi` in the repository

```bash
rg '\.fetchPypi'
```

Note that there are false positives (e.g. when used in overlays) and this also might not include all usages (let inherit)

### PR DESCRIPTION
**Pull request by figsoda**
###### Description of changes

closes #233252

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/python-aliases.nix [hunk 1]
```diff
@@ -104,6 +104,7 @@ mapAliases ({
   face_recognition_models = face-recognition-models; # added 2022-10-15
   fake_factory = throw "fake_factory has been removed because it is unused and deprecated by upstream since 2016."; # added 2022-05-30
   faulthandler = throw "faulthandler is built into ${python.executable}"; # added 2021-07-12
+  inherit fetchPypi;
```
Comment (wegank) [inline pkgs/top-level/python-aliases.nix:107]: I would actually prefer a deprecation notice here. Maybe that's too unfriendly?

Comment (figsoda) [reply pkgs/top-level/python-aliases.nix:107]: as in `throw` or `warn`? I think adding a warning here makes sense

Comment (wegank) [reply pkgs/top-level/python-aliases.nix:107]: As in `throw`, I mean.

Comment (figsoda) [reply pkgs/top-level/python-aliases.nix]: I think we might want to wait a little longer before converting this to `throw`

**Review by wegank — APPROVED** [review]

**Comment by figsoda** [comment]
fixed merge conflicts

File: pkgs/top-level/python-aliases.nix [hunk 2]
```diff
@@ -1,4 +1,4 @@
-lib: self: super:
+{ lib, fetchPypi }: self: super:
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-aliases.nix:1]:
```suggestion
{ lib, pkgs }: self: super:
```
if we add any other packages here in the future

Comment (SuperSandro2000) [reply pkgs/top-level/python-aliases.nix:1]: Honestly I am not sure and if it breaks splicing. I don't know if we have a predecessor for this. Maybe it is enough to use a throw like the other packages we move to pkgs.

Comment (FRidh) [reply pkgs/top-level/python-aliases.nix:1]: The package set should already be part of `super`, so `super.pkgs.fetchPypi`.

Comment (figsoda) [reply pkgs/top-level/python-aliases.nix]: good point, updated the PR

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**